### PR TITLE
Improved zip reports

### PR DESCRIPTION
**Pull request by JuliaSoboleva**
* Fixed bug when receipts could have `customOrderId` in `0, 1, 2...` format (described [here](https://trello.com/c/fBAl6MSW/348-fix-a-big-in-which-the-sorting-order-can-be-skewed-for-existing-users) )
* Fixed bug when generated reports which were exported to Google Drive have wrong name and have no extension
* Changed `ZIP - Receipt Files with Metadata` report behavior (now it also saves PDF attachments) 
* Added new report type `ZIP - Receipt Files` which saves all attachments in zip without stamping

### CODE REVIEW

File: app/src/main/java/co/smartreceipts/android/receipts/editor/ReceiptCreateEditFragment.java [hunk 1]
```diff
@@ -692,7 +691,23 @@ private boolean isNewReceipt() {
...
+                        TimeUnit.MINUTES.toSeconds(calendar.get(Calendar.MINUTE)) +
+                        calendar.get(Calendar.SECOND);
+        calendar.setTime(dateBox.getDate());
+        calendar.setTimeZone(TimeZone.getDefault());
+        calendar.set(Calendar.HOUR_OF_DAY, 0);
+        calendar.set(Calendar.MINUTE, 0);
+        calendar.set(Calendar.SECOND, 0);
+        calendar.set(Calendar.MILLISECOND, 0);
+
+        final Date newReceiptDate = new Date(calendar.getTimeInMillis() + currentSecondsElapsedToday);
```
Comment (wbaumann) [inline app/src/main/java/co/smartreceipts/android/receipts/editor/ReceiptCreateEditFragment.java]: What's the purpose of this? Does the `dateBox` not include the HOURS/MINUTES/SECONDS?

File: app/src/main/java/co/smartreceipts/android/receipts/editor/ReceiptCreateEditFragment.java [hunk 2]
```diff
@@ -692,7 +691,23 @@ private boolean isNewReceipt() {
 
     private void saveReceipt() {
 
-        if (presenter.checkReceipt(dateBox.getDate())) {
+        // Note: we're saving date that was picked by user (without time information) + currentSecondsElapsedToday to create ordering
+        final Calendar calendar = Calendar.getInstance();
+        final long currentSecondsElapsedToday =
+                TimeUnit.HOURS.toSeconds(calendar.get(Calendar.HOUR_OF_DAY)) +
+                        TimeUnit.MINUTES.toSeconds(calendar.get(Calendar.MINUTE)) +
+                        calendar.get(Calendar.SECOND);
```
Comment (wbaumann) [inline app/src/main/java/co/smartreceipts/android/receipts/editor/ReceiptCreateEditFragment.java]: Should this include `millis`?

File: app/src/main/java/co/smartreceipts/android/utils/IntentUtils.java [hunk 3]
```diff
@@ -103,23 +103,29 @@ public static Intent getSendIntent(@NonNull Context context, @NonNull List<File>
         Preconditions.checkNotNull(context);
         Preconditions.checkNotNull(files);
 
+        // need to use Intent.ACTION_SEND_MULTIPLE even if files.size == 1 because of Google Drive bug
+        // see: https://stackoverflow.com/questions/35382474/google-drive-changes-file-name-to-intent-extra-subject-when-sharing-a-file-via-i
+        final Intent sentIntent = new Intent(Intent.ACTION_SEND_MULTIPLE);
+
         if (files.size() == 1) {
-            return getSendIntent(context, files.get(0));
```
Comment (wbaumann) [inline app/src/main/java/co/smartreceipts/android/utils/IntentUtils.java]: Can we delete `getSendIntent`  now? Or do others still use it?

File: app/src/main/java/co/smartreceipts/android/workers/EmailAssistant.java [hunk 4]
```diff
@@ -138,42 +138,22 @@ public void emailTrip(@NonNull EnumSet<EmailOptions> options) {
         attachmentWriter.execute(trip);
     }
 
-    public void onAttachmentsCreated(File[] attachments) {
+    private void onAttachmentsCreated(File[] attachments) {
```
Comment (wbaumann) [inline app/src/main/java/co/smartreceipts/android/workers/EmailAssistant.java]: Unrelated, but we'll want to clean up this class at some point :)

File: app/src/main/java/co/smartreceipts/android/utils/IntentUtils.java (hunk 3 above)
Comment (JuliaSoboleva) [reply app/src/main/java/co/smartreceipts/android/utils/IntentUtils.java]: As I see, it's still used

File: app/src/main/java/co/smartreceipts/android/receipts/editor/ReceiptCreateEditFragment.java (hunk 2 above)
Comment (JuliaSoboleva) [reply app/src/main/java/co/smartreceipts/android/receipts/editor/ReceiptCreateEditFragment.java]: This code will be removed :)

**Pull request merged**